### PR TITLE
Create a high-level CCD client

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-810

### Change description ###

Create a high-level CCD client, for marking the awaiting of payment DCN processing as completed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+@Service
+public class CcdClient {
+
+    public static final Logger log = LoggerFactory.getLogger(CcdClient.class);
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: public?

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: Right, well spotted. Will change.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        CcdAuthenticator authenticator = authenticatorFactory.createForJurisdiction(jurisdiction);
+        String caseTypeId = service.toUpperCase() + "_ExceptionRecord";
+        String eventId = COMPLETE_AWAITING_DCN_PROCESSING_EVENT_ID;
+
+        StartEventResponse startEventResponse =
+            startEvent(authenticator, jurisdiction, caseTypeId, exceptionRecordCcdId, eventId);
+
+        Event event = Event
+            .builder()
+            .summary("Complete payment DCN processing")
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: should this be in past tense?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        this.ccdApi = ccdApi;
+        this.authenticatorFactory = authenticatorFactory;
+    }
+
+    public void completeAwaitingDcnProcessing(
+        String exceptionRecordCcdId,
+        String service,
+        String jurisdiction
+    ) {
+        log.info("Completing awaiting payment DCN processing");
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: maybe log exception record id etc

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: Good idea. I'll add the service, as well.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java (hunk 2 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: No, all the events are described like this.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            startEvent(authenticator, jurisdiction, caseTypeId, exceptionRecordCcdId, eventId);
+
+        Event event = Event
+            .builder()
+            .summary("Complete payment DCN processing")
+            .id(startEventResponse.getEventId())
+            .build();
+
+        CaseDataContent caseDataContent = CaseDataContent.builder()
+            .data(ImmutableMap.of(AWAITING_DCN_PROCESSING_FIELD_NAME, "No"))
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: is there a constant for `No` in this project?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/client/CompleteAwaitingDcnProcessingTest.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+        verify(ccdApi).startEventForCaseWorker(
+            userToken,
+            serviceToken,
+            userId,
+            jurisdiction,
+            expectedCaseTypeId,
+            exceptionRecordId,
+            COMPLETE_AWAITING_PROCESSING_EVENT_ID
+        );
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/client/CompleteAwaitingDcnProcessingTest.java]: you _could_ just put those in mocking stage

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java (hunk 4 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java]: There's no such constant in this project. This is the only occurrence.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/client/CompleteAwaitingDcnProcessingTest.java (hunk 5 above)
Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/client/CompleteAwaitingDcnProcessingTest.java]: I think it's most explicit this way.

**Pull request merged**